app/containers/wallet/staking/StakingDashboardPage.js

```diff
@@ -880,26 +894,31 @@ export default class StakingDashboardPage extends Component<Props> {
     for (let i = startEpoch; i < endEpoch; i++) {
       if (historyIterator < history.length && i === history[historyIterator][0]) {
         // exists a reward for this epoch
+        const poolHash = history[historyIterator][2];
         const nextReward = history[historyIterator][1];
         amountSum = amountSum.joinAddMutable(nextReward);
         totalRewards.push({
           name: i,
           primary: getNormalized(amountSum.getDefaultEntry()).toNumber(),
+          poolName: getMiniPoolInfo(poolHash),
         });
         perEpochRewards.push({
           name: i,
           primary: getNormalized(nextReward.getDefaultEntry()).toNumber(),
+          poolName: getMiniPoolInfo(poolHash),
         });
         historyIterator++;
       } else {
         // no reward for this epoch
         totalRewards.push({
           name: i,
           primary: getNormalized(amountSum.getDefaultEntry()).toNumber(),
+          poolName: '', // TODO: Not sure if this is OK to be empty
```

Comment:
TODO? I guess this is fine unless you want to represent it by null/undefined instead

Reply:
Great! Yeah I think empty string is fine. I will remove the TODO and merge